n-puzzle/src/n_puzzle/generator.rs

```diff
@@ -1,8 +1,47 @@
-use anyhow::Result;
+use anyhow::{Result, anyhow};
+use rand::seq::SliceRandom;
 use super::Puzzle;
+use super::pos::Pos;
```

Comment:
```suggestion
use super::{Puzzle, Pos};
```

細かい話ですが、こちらこのようにまとめることができるので、まとめるのはいかがでしょう？
このようにできるのは、n_puzzle.rsで`pub use pos::Pos;`としているためです！